<launch>

<node pkg="sr_slam" type="test_vro_img" name="test_vro_img" output="screen">
  <!-- vro parameters -->
  <param name="img_dir"        value="/media/work/work/data/realsense/Q_desk_loop"/> <!-- Q_desk_loop -->
  <param name="img_dir"        value="/media/work/work/data/realsense/etas_4f_down_swing_slow_01"/> <!-- Q_desk_loop -->
  <param name="img_dir"        value="/home/davidz/work/data/real_sense/03_14_2017_etas_xytable"/> <!-- Q_desk_loop -->
  <param name="img_dir"        value="/home/davidz/work/data/real_sense/R200"/>
 
  <param name="img_from"                          value="104" />
  <param name="img_to"                          value="100" />
  <param name="show_point_cloud"                  value="true"/>
  <param name="config/store_pointclouds"      value="true"/> 
  <!-- sift parameters -->
  <param name="sift_num_features"                   value="0" /> <!-- number of returned sift features -->
  <param name="sift_octave_layers"                  value="5" /> <!-- number octave layers -->
  <param name="sift_contrast_threshold"             value="0.08" /> <!-- 0.04 initial threshold for keypint detection -->
  <param name="siftgpu_contrast_threshold"          value="0.0018" /> <!-- 0.0018 0.0008 initial DOG threshold for siftgpu -->
  <!-- param name="siftgpu_edge_threshold"              value="10." /--> <!-- edge threshold for siftgpu  -->

  <!-- Visual Features, to activate GPU-based features see CMakeLists.txt -->
  <param name="config/feature_detector_type"         value="SIFT"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/feature_extractor_type"        value="SIFT"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/min_keypoints"                 value="20"/><!-- Extract no less than this many ... -->
  <param name="config/min_matches"                   value="12"/> <!-- Minimal number of matched features before RANSAC 29? -->
  <param name="config/max_keypoints"                 value="10000"/> <!-- maximum number of key points -->
  <param name="config/nn_distance_ratio"             value="0.7"/> <!-- 0.5/ 0.7 for SIFT, 0.95 for SIFTGPU Feature correspondence is valid if distance to nearest neighbour is smaller than this parameter times the distance to the 2nd neighbour. This needs to be 0.9-1.0 for SIFTGPU w/ FLANN, since SIFTGPU Features are normalized-->

  <param name="config/max_dist_for_inliers"          value="0.05"/> <!-- 0.05 distance for matches to be considered inliers by ransac-->
  <param name="config/ransac_iterations"             value="10000" /> <!-- 200 -->
  <param name="config/b_ransac_fixed_iter"           value="false"/> <!-- weather to use fixed   -->

  <!-- vro strategy -->
  <param name="config/vro_strategy"                 value="vro_my" /> <!-- " VRO strategy: VRO_ori, VRO_my, VRO_plane" -->

  <param name="vro_display_matches"                 value="true" /> <!-- weather to display the matched points -->

 </node>
</launch>
